fastplotlib/graphics/mesh.py

```diff
@@ -0,0 +1,172 @@
+        pygfx_cmap = pygfx.TextureMap(cmap)
+    elif isinstance(cmap, (str, dict)):
+        pygfx_cmap = pygfx.cm.create_colormap(get_cmap(cmap))
+    else:
+        map = np.asarray(cmap)
+        if map.ndim == 2:  # 1D plus color
+            pygfx_cmap = pygfx.cm.create_colormap(cmap)
+        else:
+            tex = pygfx.Texture(map, dim=map.ndim - 1)
+            pygfx_cmap = pygfx.TextureMap(tex)
+
+    return pygfx_cmap
+
+
+class MeshGraphic(Graphic):
+    _features = {
+        "positions": VertexPositions,
+        "indices": BufferManager,
+        "mapcoords": (BufferManager, None),
+        "colors": (VertexColors, UniformColor),
+    }
+
+    def __init__(
+        self,
+        positions: Any,
+        indices: Any,
+        colors: str | np.ndarray | Sequence = "w",
+        mapcoords: Any = None,
+        cmap: str = None,
+        isolated_buffer: bool = True,
+        **kwargs,
+    ):
+        """
+        Create a mesh Graphic.
+
+        Parameters
+        ----------
+        positions: array-like
+            The 3D positions of the vertices.
+
+        indices: array-like
+            The indices into the positions that make up the triangles. Each 3
+            subsequent indices form a triangle.
+
+        colors: str, array, or iterable, default "w"
+            A uniform color, or the per-position colors.
+
+        mapcoords: array-like
+            The per-position coordinates to which to apply the colormap (a.k.a. texcoords).
+            These can e.g. be some domain-specific value, mapped to [0..1].
+            If ``mapcoords`` and ``cmap`` are given, they are used instead of ``colors``.
+
+        cmap: str, optional
+            Apply a colormap to the mesh, this overrides any argument passed to
+            "colors". For supported colormaps see the ``cmap`` library
+            catalogue: https://cmap-docs.readthedocs.io/en/stable/catalog/
+            Both 1D and 2D colormaps are supported, though the mapcoords has to match the dimensionality.
+
+        **kwargs
+            passed to :class:`.Graphic`
+
+        """
+
+        super().__init__(**kwargs)
+
+        if isinstance(positions, VertexPositions):
+            self._positions = positions
+        else:
+            self._positions = VertexPositions(
+                positions, isolated_buffer=isolated_buffer, property_name="positions"
+            )
+
+        if isinstance(positions, BufferManager):
+            self._indices = indices
+        else:
+            self._indices = BufferManager(
+                indices, isolated_buffer=isolated_buffer, property_name="indices"
+            )
+
+        if mapcoords is None:
+            self._mapcoords = None
+        elif isinstance(mapcoords, BufferManager):
+            self._mapcoords = mapcoords
+        else:
+            self._mapcoords = mapcoords = BufferManager(
+                mapcoords, isolated_buffer=isolated_buffer, property_name="mapcoords"
+            )
```

Comment:
I think changing texcoords would be done in-conjunction with changes to positions, indices, or the cmap/image so we can just make this a simple property instead of a `BufferManager`